test/torch/nn/test_rnn.py

```diff
@@ -0,0 +1,182 @@
+import pytest
+import torch as th
+import syft as sy
+from syft.frameworks.torch.nn import RNN
+from syft.frameworks.torch.nn import GRU
+from syft.frameworks.torch.nn import LSTM
+
+
+@pytest.mark.parametrize("nonlinearity", ["tanh", "relu"])
+def test_simple_rnn(nonlinearity, hook, workers):
+    """
+    Testing RNN modules with MPC
+    """
+
+    bob = workers["bob"]
+    alice = workers["alice"]
+    crypto_prov = sy.VirtualWorker(hook, id="crypto_prov")
+
+    # model hyperparameters
+    batch_size = 4
+    input_size = 3
+    hidden_size = 5
+    seq_len = 6
+    num_layers = 2
+    bidirectional = True
+    directions = 2 if bidirectional else 1
+
+    x = (
+        th.randn(batch_size, seq_len, input_size)
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+    model = (
+        RNN(
+            input_size=input_size,
+            hidden_size=hidden_size,
+            num_layers=num_layers,
+            nonlinearity=nonlinearity,
+            batch_first=True,
+            bidirectional=bidirectional,
+        )
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+
+    out, h = model(x)
+    assert out.get().float_precision().shape == th.Size(
+        [batch_size, seq_len, directions * hidden_size]
+    )
+    assert h.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
+
+    h0 = (
+        th.randn(batch_size, directions * num_layers, hidden_size)
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+    out, h = model(x, h0)
+    assert out.get().float_precision().shape == th.Size(
+        [batch_size, seq_len, directions * hidden_size]
+    )
+    assert h.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
+
+
+def test_gru(hook, workers):
```

Comment:
Is this test exactly like `test_simple_rnn` but for GRU? If so, it would be better to add the cell type as a parameter at @pytest.mark.parametrize

Reply:
Not exactly equal, as GRU and LSTM don't have the arg `nonlinearity` that RNN has. I prefer to keep them separate or we will have to iterate twice for GRU and LSTM (by using two values for `nonlinearity`) with no need to do so.

Reply:
hmmm, I think you could just use one argument (something like 'model-type') and you could set it to: gru, lstm, rnn-tanh or rnn-relu. Internally you could deal with the specificity of each model (aka setting the nonlinearity when needed) What do you think?

I think this is better than having 3 different tests that are not actually that different.

Reply:
I see. Ok, I will do like that